listenbrainz/webserver/views/playlist_api.py

```diff
@@ -286,6 +286,10 @@ def create_playlist():
         get("extension", {}).get(PLAYLIST_EXTENSION_URI, {}).\
         get("collaborators", [])
 
+    # Don't allow creator to also be a collaborator
+    if user["musicbrainz_id"] in collaborators:
```

Comment:
If someone puts the a collaborator name in the list more than once this will only remove one of them. We should also make this list unique before filtering and before passing it to the db method

Reply:
Good catch, thanks

Reply:
the python way of doing this is `collaborators = list(set(collaborators))`, instead of going through dict.

Reply:
> the python way of doing this is `collaborators = list(set(collaborators))`, instead of going through dict.

I just realized `list(set(…` doesn't conserve the original list order while `list(dict.fromkeys(…` does.
I don't think that's of much importance considering we always overwrite the collaborators, but just wanted to check if I could be missing some other reason why order might be important…

Reply:
Why don't we make a habit of keeping the list sorted? In which case, we just add a sorted() call on the outside and you're done.

Reply:
in that case we should probably sort when pulling out of the database rather than when inserting - we have no ordering on the SELECT statement so it's not guaranteed that this would come out ordered anyway.

Reply:
Created a ticket: https://tickets.metabrainz.org/projects/LB/issues/LB-807